quickwit/quickwit-serve/src/elasticsearch_api/bulk_v2.rs

```diff
@@ -148,19 +165,69 @@ pub(crate) async fn elastic_bulk_ingest_v2(
                     None,
                 )
             })?;
-        for es_doc_id in es_doc_ids {
+        let index_id = success
+            .index_uid
+            .map(|index_uid| index_uid.index_id)
+            .expect("`index_uid` should be a required field");
+
+        // Populate the response items with one error per parse failure.
+        for parse_failure in success.parse_failures {
+            // Since the generated doc UIDs are monotonically increasing, and inserted in order, we
+            // can find doc handles using binary search.
+            assert!(
```

Comment:
Let's move this assert out of the loop?

Reply:
No, the `Vec` of `DocHandle` is built locally and doesn't go through gRPC.